sdk/src/main/java/com/hedera/hashgraph/sdk/AccountAllowanceDeleteTransaction.java

```diff
@@ -0,0 +1,172 @@
+package com.hedera.hashgraph.sdk;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hedera.hashgraph.sdk.proto.CryptoDeleteAllowanceTransactionBody;
+import com.hedera.hashgraph.sdk.proto.CryptoServiceGrpc;
+import com.hedera.hashgraph.sdk.proto.SchedulableTransactionBody;
+import com.hedera.hashgraph.sdk.proto.Transaction;
+import com.hedera.hashgraph.sdk.proto.TransactionBody;
+import com.hedera.hashgraph.sdk.proto.TransactionResponse;
+import io.grpc.MethodDescriptor;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class AccountAllowanceDeleteTransaction extends com.hedera.hashgraph.sdk.Transaction<AccountAllowanceDeleteTransaction> {
+    private final List<HbarAllowance> hbarAllowances = new ArrayList<>();
+    private final List<TokenAllowance> tokenAllowances = new ArrayList<>();
+    private final List<TokenNftAllowance> nftAllowances = new ArrayList<>();
+    // <ownerId, <tokenId, index>>
+    private final Map<AccountId, Map<TokenId, Integer>> nftMap = new HashMap<>();
+
+    public AccountAllowanceDeleteTransaction() {
+    }
+
+    AccountAllowanceDeleteTransaction(
+        LinkedHashMap<TransactionId, LinkedHashMap<AccountId, com.hedera.hashgraph.sdk.proto.Transaction>> txs
+    ) throws InvalidProtocolBufferException {
+        super(txs);
+        initFromTransactionBody();
+    }
+
+    AccountAllowanceDeleteTransaction(com.hedera.hashgraph.sdk.proto.TransactionBody txBody) {
+        super(txBody);
+        initFromTransactionBody();
+    }
+
+    private void initFromTransactionBody() {
+        var body = sourceTransactionBody.getCryptoDeleteAllowance();
+        for (var allowanceProto : body.getCryptoAllowancesList()) {
+            hbarAllowances.add(HbarAllowance.fromProtobuf(allowanceProto));
+        }
+        for (var allowanceProto : body.getTokenAllowancesList()) {
+            tokenAllowances.add(TokenAllowance.fromProtobuf(allowanceProto));
+        }
+        for (var allowanceProto : body.getNftAllowancesList()) {
+            getNftSerials(
+                AccountId.fromProtobuf(allowanceProto.getOwner()),
+                TokenId.fromProtobuf(allowanceProto.getTokenId())
+            ).addAll(allowanceProto.getSerialNumbersList());
+        }
+    }
+
+    public AccountAllowanceDeleteTransaction revokeHbarAllowance(AccountId ownerAccountId) {
```

Comment:
Hmm, I think this API is fine, but I am wondering what your thoughts are on the following:
`revokeHbarAllowances` - since technically this revokes _all_ allowances granted by the owner
`revokeAllHbarAllowances` - perhaps we should be explicit here?
`deleteHbarAllowances` - to keep in line with the name of the transaction in both the SDK and the protobufs
`deleteAllHbarAllowances` - a combination of both previous points
This would apply to the other two allowances as well.